sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/README.md

```diff
@@ -0,0 +1,213 @@
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+
+<!-- CLIENT COMMON BAR -->
+[Client options](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#configuring-service-clients-using-clientoptions) |
+[Accessing the response](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#accessing-http-response-details-using-responset) |
+[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#consuming-long-running-operations-using-operationt) |
+[Handling failures](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#reporting-errors-requestfailedexception) |
+[Diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md) |
+[Mocking](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#mocking) |
+[Client lifetime](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/)
+<!-- CLIENT COMMON BAR -->
+
+## Examples
+
+The Azure.AI.Language.QuestionAnswering client library provides both synchronous and asynchronous APIs.
+
+The following examples show common scenarios using the `client` [created above](#create-questionansweringclient).
+
+### Ask a question
+
+The only input required to a ask a question using a knowledgebase is just the question itself:
+
+```C# Snippet:QuestionAnsweringClient_QueryKnowledgebase
+KnowledgebaseQueryOptions options = new KnowledgebaseQueryOptions("How long should my Surface battery last?");
+
+Response<KnowledgebaseAnswers> response = client.QueryKnowledgebase("FAQ", options);
+
+foreach (KnowledgebaseAnswer answer in response.Value.Answers)
+{
+    Console.WriteLine($"({answer.ConfidenceScore:P2}) {answer.Answer}");
+    Console.WriteLine($"Source: {answer.Source}");
+    Console.WriteLine();
+}
+```
+
+You can set additional properties on `QuestionAnsweringClientOptions` to limit the number of answers, specify a minimum confidence score, and more.
+
+### Ask a follow-up question
+
+If your knowledgebase is configured for [chit-chat][questionanswering_docs_chat], you can ask a follow-up question provided the previous question-answering ID and, optionally, the exact question the user asked:
+
+```C# Snippet:QuestionAnsweringClient_Chat
+KnowledgebaseQueryOptions options = new KnowledgebaseQueryOptions("How long should charging take?")
+{
+    Context = new KnowledgebaseAnswerRequestContext(previousAnswer.Id.Value)
```

Comment:
I had named it in hopes that it was an obvious reference to the previous answer, but since that wasn't clear I will change it somehow to clarify. It was supposed to be like a follow-up from the previous sample. Perhaps I can simulate something of that to feed into this question.